src/aks-preview/azext_aks_preview/_validators.py

```diff
@@ -585,3 +585,11 @@ def validate_azure_keyvault_kms_key_id(namespace):
         segments = key_id[len(https_prefix):].split("/")
         if len(segments) != 4 or segments[1] != "keys":
             raise InvalidArgumentValueError(err_msg)
+
+
+def validate_enable_custom_ca_trust(namespace):
+    """Validates Custom CA Trust can only be used on Linux."""
+    if namespace.enable_custom_ca_trust is not None and namespace.enable_custom_ca_trust:
+        if namespace.os_type is not None and namespace.os_type != "Linux":
```

Comment:
Don't worry, you could compare `None` with a `str` in Python.
```suggestion
        if namespace.os_type != "Linux":
```

Reply:
I'm having some problems with this validator here - we only want to allow enabling this on Linux nodepools, so for enable_custom_ca_trust I added validate_enable_custom_ca_trust.

To check if its linux we need access to os_type (validator code [here](https://github.com/Azure/azure-cli-extensions/pull/4782/files#diff-69ae299f91087897e4f8a2f6dcdb372c2d098597c89cd41a40c3558cd901a3eeR590)). However we allow use of custom ca flag for ```aks nodepool update``` - how can we get access to os_type in aks nodepool update flow? For now we are using has_attr to avoid null ptr in update flow.